server/controllers/finance/debtorGroups.js

```diff
@@ -0,0 +1,177 @@
+/**
+* The Debtor Groups Controllers
+*
+* @module finance/debtorGroups
+*
+* @desc This module is responsible for handling all crud operations relatives
+* to debtor groups, and relatives functions
+*
+* @required q
+* @required lib/db
+* @required lib/guid
+*
+* @todo implements the creation of a Debtor Group
+*/
+
+'use strict';
+
+var q  = require('q'),
+    db = require('../../lib/db'),
+    guid = require('../../lib/guid');
+
+/** Get debtor group details */
+exports.detail = getDetail;
+
+/** Get the list of debtor group */
+exports.list = getList;
+
+/** [HTTP API ENDPOINT] Get debtor groups invoices list */
+exports.getInvoices = getInvoices;
+
+/** [SERVER SIDE FUNCTION] Get debtor groups invoices list */
+exports.fetchInvoices = fetchInvoices;
+
+
+/**
+* GET /debtor_groups/:uuid
+*
+* @function getDetail
+*
+* @desc This function is responsible for retrieving details of a debtor group
+*/
+function getDetail(req, res, next) {
+  var uuid = req.params.uuid;
+
+  var debtorDetailsQuery =
+    'SELECT uuid, name, account_id, location_id, phone, email, note, locked ' +
+      'max_credit, is_convention, price_list_uuid ' +
+    'FROM debitor_group ' +
+    'WHERE uuid = ?';
+
+  db.exec(debtorDetailsQuery, [uuid])
+  .then(function (rows) {
+    if (!rows.length) { return next(new req.codes.ERR_NOT_FOUND()); }
+    res.status(200).json(rows[0]);
+  })
+  .catch(next)
+  .done();
+
+}
+
+/**
+* GET /debtor_groups/
+* GET /debtor_groups/?with_locked={1|0}
```

Comment:
The query string parameter should be called `locked` to match the field in the database, not `with_locked`.

I think this function doesn't cover all cases.
 1. `GET /debtor_groups`  should return all debtor groups, without consideration of the locked parameter.  This is correctly handled.
 2. `GET /debtor_groups?locked=1` should ONLY return debtor groups with locked = 1.  This is correctly handled.
 3. `GET /debtor_groups?locked=0` should ONLY return debtor groups with locked = 0.  In this case, you will execute the following query:

 ```js
query = 'SELECT uuid, name, locked, account_id FROM debitor_group ';
```

 ... which will return the same thing as `GET /debtor_groups`.  There should be a filter on the `locked` field.